### ISSUE TITLE
Does it remove unused icons?

### ISSUE THREAD

**Issue by Tobeyforce** [patrickxchong/nuxt-vuetify-purgecss#11]
Hey,
I'm attempting to use purgecss with Nuxt as well.
Does the current config for this project removed unused icons from mdi? This seems like the biggest bottleneck so far but having found some config settings I still can't get the 300kb icon batch to be removed. 
Using the nuxt@purgecss and adding this to the paths:
      './node_modules/vuetify/dist/*.js',
      './node_modules/vuetify/dist/*.css',
      './node_modules/vuetify/src/**/*.ts',

**Comment by patrickxchong** [comment]
Hey @Tobeyforce are you using the default 'mdi' icon setting? That unfortunately includes the entire icon font. To optimise for production you should use the mdi SVGs in from @mdi/js (refer to https://vuetifyjs.com/en/features/icon-fonts/#material-design-icons-js-svg)

**Comment by Tobeyforce** [comment]
Hey,
That's exactly what I did before but because all vuetify components use icons constantly I had to manually insert the icons (e.g v-select, v-combobox) which just made it unrealistic from a time perspective to always write extra code. It worked great with v-checkbox because it has props for it but not all components make it easy. For example, v-select uses v-menu which uses v-checkbox which needs its own icons. Is there a way to do these manual imports and make sure they are automatically used by vuetify without explicitly adding them?

**Comment by patrickxchong** [comment]
Unfortunately that's how it is with Vuetify. I've not been able to find any good workarounds other than building my own components/reskinning 3rd party components to fit the Vuetify UI. At this point I won't recommend Vuetify to anyone because it's just so hard to optimise, and I don't plan to use it for any of my future projects. There are other libraries that don't seem to have performance issues such as https://github.com/ElemeFE/element or https://github.com/vueComponent/ant-design-vue. (though I've not used them yet because I've just been using tailwind to style and writing my own logic for most of my projects these days)

**Comment by Tobeyforce** [comment]
Thanks for the reply, really appreciate it.
It baffles me that this isn't done out of the box considering the size of it. They're just like "import stuff manually".. and then you notice all default icons for the components are gone and you're left wondering what to do. Nah, lets work on pointless features instead.. ;)
I might check out those libraries in the future, this is what I did though:
1. Installed the 'md' icons locally from https://www.npmjs.com/package/material-design-icons-iconfont
2. Installed mdi/js 
Basically, I use the 'md' icons for default & whenever I need some more specific icons that aren't present I use the treeshaken 'mdi' ones. This netted me around 80kb of icons from the 'md' library, but it's a considerable improvement compared to 330 or so.

**Comment by patrickxchong** [comment]
Ahh that's a neat trick. Thanks for sharing that @Tobeyforce! Yeahh, Vuetify is popular and has a lot of potential, but unfortunately they aren't focusing enough on performance+developer experience. Will close this issue.
